### DISCUSSION TITLE
Relatively large memory footprint of the initial project (57.9MB)

### DISCUSSION THREAD

**Discussion by MikeChongCan** [wailsapp/wails#777] [General]
<img width="787" alt="Screen Shot 2021-08-14 at 1 25 20 PM" src="https://user-images.githubusercontent.com/2396817/129435102-c722f4da-400f-4f68-9065-d8c8e3b5abdd.png">

Hello, this project is amazing and I love it!

After building the project, I found the memory footprint it quite high. Any way to fix it?

**Comment by leaanthony** [comment]
Well, high is relative 😃 if you can find a way I'll always be happy to accept a PR!

**Reply by leaanthony** [reply]
Also note that the bulk of this will be fixed overhead. It sure beats an electron project.

**Comment by mgutz** [comment]
I'm observing the same on Linux with the current Wails version. I created a simple petite-vue counter app in Wails and compared the memory usage against  VS Code with no files open. Surprisingly, the trivial Wails app (production build) used more memory.

This might just come down to Chromium being more resource friendly than Webview. Is the tradeoff for a smaller binary worth it?

**Comment by kazzkiq** [comment]
@mgutz it looks like a Linux WebKit issue, all projects that use webviews for UI face it, not only Wails.

Check this table comparing memory usage from a simple app ([source here](https://github.com/Elanis/web-to-desktop-framework-comparison/blob/main/README.md#memory-usage)):

### Memory Usage  

|  | [Electron](https://github.com/electron/electron) | [NW.JS](https://github.com/nwjs/nw.js) | [Tauri](https://github.com/tauri-apps/tauri) | [NodeGui](https://github.com/nodegui/nodegui) | [Neutralino](https://github.com/neutralinojs/neutralinojs) | [Wails](https://github.com/wailsapp/wails) | [Flutter](https://github.com/flutter/flutter) | [.Net MAUI](https://github.com/dotnet/maui) | [.Net MAUI w/ Vue](https://github.com/dotnet/maui) |
|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
| ***Windows (x64)*** | ≈67MB (Debug) => ≈88MB (Release) | ≈102MB (Debug) => ≈40MB (Release) | ≈41MB (Debug) => ≈25MB (Release) | ≈123MB (Debug) | ≈69MB (Debug) | ≈49MB (Debug) => ≈28MB (Release) | ≈274MB (Debug) | N/A<sup>1</sup>| N/A<sup>2</sup>|
| ***MacOS (x64)*** | ≈66MB (Debug) => ≈65MB (Release) | ≈53MB (Debug) => ≈82MB (Release) | ≈50MB (Debug) => ≈29MB (Release) | ≈107MB (Debug) | ≈77MB (Debug) => ≈28MB (Release) | ≈65MB (Debug) => ≈32MB (Release) | ≈186MB (Debug) | N/A<sup>1</sup>| N/A<sup>2</sup>|
| ***Linux (x64)*** | ≈166MB (Debug) => ≈165MB (Release) | ≈96MB (Debug) | ≈116MB (Debug) => ≈83MB (Release) | ≈163MB (Debug) | ≈198MB (Debug) => ≈216MB (Release) | ≈220MB (Debug) => ≈195MB (Release) | ≈399MB (Debug) | N/A<sup>1</sup>| N/A<sup>2</sup>|

Wails perform better than most options on the list and pretty close to Tauri in all platforms, with only Linux behaving oddly and using huge amounts of memory for them all.

But certainly there is room for improvement, since Tails managed to drop their memory usage on Linux quite a bit.

**Reply by leaanthony** [reply]
Oh wow, how on earth do we get such wildly different figures than Tauri for Linux??? Is that a Go issue?
